### ISSUE TITLE
[es/snapshot.restore] Failed to decode Response. Missing required property 'RestoreResponse.snapshot'

### ISSUE THREAD

**Issue by ACAA06** [elastic/elasticsearch-java#891]
### Java API client version

8.14.3

### Java version

11

### Elasticsearch Version

8.15

### Problem description

Logs:
`co.elastic.clients.transport.TransportException: node: https://************:9200/, status: 200, [es/snapshot.restore] Failed to decode response`

> Java code which creates the snapshot:

            RestoreRequest.Builder restoreRequestBuilder = new RestoreRequest.Builder();
            restoreRequestBuilder.repository(repoName);
            restoreRequestBuilder.snapshot(snapshot);
            restoreRequestBuilder.indices(indices);
            
            elasticsearchClient.snapshot().restore(restoreRequestBuilder.build());

The snapshot is successfully created on Elasticsearch, but the code fails to parse the response from the server, which simply returns:

>  
            {
              "accepted": true
            }

But the sdk requires a non-null snapshot:

>    
            private RestoreResponse(Builder builder) {
                  this.snapshot = (SnapshotRestore) ApiTypeHelper.requireNonNull(builder.snapshot, this, "snapshot");
              }

Can someone help me with this please? Am i doing something wrong here?

**Comment by l-trotta** [comment]
Hello, thanks for reporting this! What you're doing seems correct, so it looks like the server returns something that we had not considered in the [API specification](https://github.com/elastic/elasticsearch-specification) used to produce the Java code; we'll fix the specification and regenerate the client. 

In the meantime, what you can do as a workaround is disable the check just for that specific call, by performing it inside a try block configured like this:

```
try (ApiTypeHelper.DisabledChecksHandle h = ApiTypeHelper.DANGEROUS_disableRequiredPropertiesCheck(true)) {
 elasticsearchClient.snapshot()
    .restore(r -> r
        .indices(indices)
        .repository(repository)
        .snapshot(snapshot));
}
```

As the method name says, it's dangerous to use because it basically ignores every check the client normally performs on non nullable properties, so only use it to make the call work (all fields in the response will be null, but in this case you don't need it).

I'll update this when the fix becomes available in the next patch, so that you can just use the API normally.

**Comment by ACAA06** [comment]
Hi, Thank you for your response.

**Comment by l-trotta** [comment]
fixed in the latest 8.15.x as well as in 8.16.0!
